File: finding_given_array_is_palindrome_or_not.java
public class Main {
    public static void main(String[] args) {
        char[] array = {'a', 'b', 'd', 'e', 'd', 'b', 'a'};
        int n = array.length;
        int i = 0;
        int j = n - 1;

        while (i <= j) {
            if (array[i] != array[j]) {
                System.out.println("The given array is not a palindrome");
                return; // exit the program
            }
            i++;
            j--;
        }

        System.out.println("The given array is a palindrome");
    }
}

File: finding_given_array_is_palindrome_or_not.py
array = ['a', 'b', 'd', 'e', 'd', 'b', 'a']
n = len(array)
i = 0
j = n - 1

while i <= j:
    if array[i] != array[j]:
        print("The given array is not a palindrome")
        break
    i += 1
    j -= 1
else:
    print("The given array is a palindrome")
